packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeForkClient.ts

```diff
@@ -88,7 +90,14 @@ Please use block number ${lastSafeBlock} or wait for the block to get ${
     cacheToDiskEnabled ? forkCachePath : undefined
   );
 
-  return { forkClient, forkBlockNumber, forkBlockTimestamp };
+  const forkBlockHash = block.hash;
+
+  assertHardhatInvariant(
+    forkBlockHash !== null,
+    "Forked block should have a hash"
+  );
```

Comment:
I think only pending blocks have a null hash, so this should be fine.

Reply:
Yes, this is a valid assumption